### PR TITLE
Add `subsync.app`

### PR DESCRIPTION
**Pull request by zx8**
After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

---

Ref: https://github.com/sc0ty/subsync/issues/88

### CODE REVIEW

File: Casks/subsync.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+cask 'subsync' do
+  version '0.15.0'
+  sha256 '0e6f54806e30db861acc88592de474e026fa54e8b1d636e52be5fa2a5e941c0d'
+
+  # github.com/sc0ty/subsync was verified as official when first introduced to the cask
+  url "https://github.com/sc0ty/subsync/releases/download/0.15/subsync-#{version}-mac-x86_64.dmg"
```
Comment (vitorgalvao) [inline Casks/subsync.rb:6]: The `0.15` needs to be `#{version.major_minor}`.

File: Casks/subsync.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  appcast 'https://github.com/sc0ty/subsync/releases.atom'
+  name 'subsync'
+  homepage 'https://subsync.online/'
+
```
Comment (vitorgalvao) [inline Casks/subsync.rb:10]: It’s missing the `app` stanza.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/subsync.rb (hunk 2 above)
Comment (gechr) [reply Casks/subsync.rb:10]: https://github.com/Homebrew/homebrew-cask/pull/84745/commits/2c9e060c674089bec997cf4aeb6251bd0610d71f

Comment (zx8) [reply Casks/subsync.rb:10]: https://github.com/Homebrew/homebrew-cask/pull/84745/commits/2c9e060c674089bec997cf4aeb6251bd0610d71f

File: Casks/subsync.rb (hunk 1 above)
Comment (zx8) [reply Casks/subsync.rb]: https://github.com/Homebrew/homebrew-cask/pull/84745/commits/acf259304541810df9b12d314f54dfce67cabfec

File: Casks/subsync.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  app 'subsync.app'
+  # shim script (https://github.com/Homebrew/homebrew-cask/issues/18809)
+  shimscript = "#{staged_path}/subsync.wrapper.sh"
+  binary shimscript, target: 'subsync'
+
+  preflight do
+    IO.write shimscript, <<~EOS
+      #!/bin/sh
+      exec #{appdir}/subsync.app/Contents/MacOS/subsync --cli "$@"
```
Comment (miccal) [inline Casks/subsync.rb:19]: I think this should be:
```
exec '#{appdir}/subsync.app/Contents/MacOS/subsync --cli' "$@"
```

Comment (zx8) [reply Casks/subsync.rb]: https://github.com/Homebrew/homebrew-cask/pull/84745/commits/e1b22c2c8dbec7bf73e84746f501aabe7d17a73b

File: Casks/subsync.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  app 'subsync.app'
+  # shim script (https://github.com/Homebrew/homebrew-cask/issues/18809)
+  shimscript = "#{staged_path}/subsync.wrapper.sh"
+  binary shimscript, target: 'subsync'
+
+  preflight do
+    IO.write shimscript, <<~EOS
+      #!/bin/sh
+      exec '#{appdir}/subsync.app/Contents/MacOS/subsync' --cli "$@"
```
Comment (miccal) [inline Casks/subsync.rb:19]: Thanks @zx8, but the trailing `'` should be after the `--cli` part, not after the `subsync` part.

Comment (zx8) [reply Casks/subsync.rb:19]: `subsync --cli` is not a single binary containing a space. The `--cli` part is a parameter to the `subsync` binary which enables CLI-mode.

Comment (miccal) [reply Casks/subsync.rb:19]: @zx8 you are correct, apologies.

**Comment by vitorgalvao** [comment]
Still missing [`zap`](https://github.com/sc0ty/subsync/issues/88#issuecomment-647169572) and [`depends_on`](https://github.com/sc0ty/subsync/issues/88#issuecomment-647227138) entries.

**Comment by zx8** [comment]
@vitorgalvao I'm not sure what you mean by missing `zap` entries? My understanding based on @sc0ty's linked comment was that `~/Library/Preferences/subsync/` should cover everything – what am I missing?

@sc0ty Before I add the `depends_on` stanza, can you confirm that the dependencies you mentioned in the linked comment ("ffmpeg or libav and pocketsphinx") do not come bundled with the DMG file? Do they need to be installed separately?

**Comment by vitorgalvao** [comment]
> My understanding based on @sc0ty's linked comment was that `~/Library/Preferences/subsync/` should cover everything – what am I missing?

Nothing. Seems I misread, as I thought there were more files.

File: Casks/subsync.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    EOS
+  end
+
+  zap trash: [
```
Comment (vitorgalvao) [inline Casks/subsync.rb:23]: If it’s only one, make it a string.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by sc0ty** [comment]
> @sc0ty Before I add the `depends_on` stanza, can you confirm that the dependencies you mentioned in the linked comment (_"ffmpeg or libav and pocketsphinx"_) do not come bundled with the DMG file? Do they need to be installed separately?

All dependencies are bundled in. I was thinking about eventual future builds made specifically for brew - with libraries installed also from brew. But that's to be consider in the future. This version should work out of the box.

File: Casks/subsync.rb (hunk 5 above)
Comment (zx8) [reply Casks/subsync.rb]: https://github.com/Homebrew/homebrew-cask/pull/84745/commits/067f184b3bf057a3416b769428d045e08d9e04dd

As a side note, would be cool if `brew cask style --fix` could detect and auto-fix this?

**Review by ran-dall — APPROVED** [review]

**Pull request merged**